wrapper/src/main/java/software/amazon/jdbc/plugin/federatedauth/WrappedFactory.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.federatedauth;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import javax.net.ssl.SSLSocketFactory;
+
+public abstract class WrappedFactory extends SSLSocketFactory {
```

Comment:
we need a better name than WrappedFactory - as wrapped conflicts with wrapper, the driver itself
Does it just wraps a SSLSocketFactory?
ExtendedSSLSocketFactory? WrappedSSLSocketFactory?

Reply:
ExtendedSSLSocketFactory it is!

Reply:
Done!